src/confluent_kafka/src/Admin.c

```diff
@@ -454,19 +455,19 @@ static PyObject *Admin_create_topics (Handle *self, PyObject *args,
                 }
 
                 if (newt->replica_assignment) {
-                        if (newt->replication_factor != -1) {
+                        if (newt->replication_factor != -1 || newt->num_partitions !=-1) {
                                 PyErr_SetString(PyExc_ValueError,
-                                                "replication_factor and "
+                                                "replication_factor/num_partitions and "
```

Comment:
@edenhill might get upset with line lengths > 80 characters, though i don't believe this is strictly a convention in this code base.